firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/analytics/AnalyticsConnectorReceiver.java

```diff
@@ -91,11 +102,22 @@ public void onMessageTriggered(int id, @Nullable Bundle extras) {
       params = new Bundle();
     }
 
-    try {
-      final String serializedEvent = BREADCRUMB_PREFIX + serializeEvent(name, params);
-      breadcrumbHandler.dropBreadcrumb(serializedEvent);
-    } catch (JSONException e) {
-      Logger.getLogger().w(Logger.TAG, "Unable to serialize Firebase Analytics event.");
+    final String origin = params.getString(EVENT_ORIGIN_KEY);
+    if (CRASH_ORIGIN.equals(origin)) {
+      if (crashOriginEventListener != null) {
```

Comment:
Funny enough, this was another reason why I was considering Set of listeners instead of just the single one. The empty loop would no-op.

I don't love the no object pattern as I find it confusing when it appears an event should be firing but there's no response anywhere. How  strongly do you feel about it in this particular place?

Reply:
Not strong enough to have you change it now. :) Just a pattern I happen to prefer that would reduce the number of times this conditional is triggered.

Reply:
In practice, though, this really should only happen once per session.

Reply:
After a little synchronous chat: we settled on this being a relatively inconsequential matter of opinion and we're leaving it as-is.